docs/spec/dns-integration.md

```diff
@@ -4,23 +4,32 @@ DNS Integration for NSM
 Specification
 -------------
 
-Network Service Mesh needs to be able to provide a workload with DNS service from Network Services Without breaking K8s DNS
+Network Service Mesh needs to be able to provide a workload with DNS resolution for Network Services without breaking the DNS resolution K8s natively provides.
+
+Overview
+--------
+
+The DNS integration capability is performed by running a DNS server co-resident with the application pod that can direct requests to multiple DNS servers. The DNS servers that should be sent requests is controlled by the NSEs that create connections into that application pod.  If the NSEs populate the DNSContext with a DNS server appropriate for this connection then additional containers in the application pod will ensure those DNS servers are consulted. The additional containers are nsm-coredns and nsm-dns-monitor.  They can be inserted directly or via the admission webhook.  The nsm-coredns container is responsible for handling the DNS resolution mechanics. The nsm-dns-monitor container is responsible for updating the configuration that the nsm-coredns container acts on based on DNS information in the connection contexts provided by the NSEs.
+
+The NSM provided SDK provides a capability to read environmental varibles to determine additional DNS servers and domains that should be added to the DNScontext.  These variables are: DNS_SEARCH_DOMAINS and DNS_SERVER_IPS.
```

Comment:
I don't see these env variables used in the sdk/ .  I do see them used in some test/applications/... but not the SDK.

The parts of the SDK I find with DNS related bits are:

[NewAddDnsConfigs](https://github.com/networkservicemesh/networkservicemesh/blob/master/sdk/endpoint/dns.go#L33)

and

[NewAddDnsConfigDstIP](https://github.com/networkservicemesh/networkservicemesh/blob/master/sdk/endpoint/dns.go#L59)

Did I miss something?

Reply:
You didn't miss anything - I explained this more in terms what the test applications does vs. what the SDK does.   I will need to fix that up.